File: leibniz.rb
#!/usr/bin/ruby
def leibnizPow(i)
  return (-1.0 ** i) / (2.0 * i + 1.0)
end

def leibniz(n)
  i = 0.0
  r = 0.0
  while i < n do
    i += 1.0
    r += leibnizPow(i)
  end
  return 4.0 * r
end

print(leibniz($*[0].to_i * 1.0))

File: leibniz.js
#!/usr/bin/node
function leibnizPow(i) {
  return Math.pow(-1, i) / (2 * i + 1);
}

function leibniz(n) {
  var r = 0;
  for (var i = 0; i < n; i++) {
    r += leibnizPow(i);
  }
  return 4 * r;
}

console.log(leibniz(parseInt(process.argv[2])));
